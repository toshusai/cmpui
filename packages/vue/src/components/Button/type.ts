export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
