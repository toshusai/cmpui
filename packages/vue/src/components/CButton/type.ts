export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
