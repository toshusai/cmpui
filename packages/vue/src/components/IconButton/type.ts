export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
