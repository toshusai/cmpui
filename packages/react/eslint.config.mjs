"use strict";
import eslintConfigPrettier from "eslint-config-prettier";
import tseslint from "typescript-eslint";
import eslintPluginImport from "eslint-plugin-import";

export default [
  ...tseslint.configs.recommended,
  {
    ignores: ["dist/*", "node_modules/*", "storybook-static/*"],
  },
  {
    plugins: { import: eslintPluginImport },
    rules: {
      "vue/multi-word-component-names": "off",
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": ["error"],
      "import/order": [
        "error",
        {
          "newlines-between": "always",
        },
      ],
    },
    languageOptions: {
      parserOptions: { parser: tseslint.parser, sourceType: "module" },
    },
  },
  eslintConfigPrettier,
];
