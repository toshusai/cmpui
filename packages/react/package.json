{
  "name": "@toshusai/cmpui",
  "version": "1.0.0-alpha.13",
  "description": "CmpUI is a component library designed for creating complex GUI applications in web browsers.",
  "main": "dist/index.mjs",
  "types": "dist/index.d.mts",
  "scripts": {
    "typecheck": "tsc -p tsconfig.ci.json",
    "test:watch": "vitest",
    "test": "vitest run",
    "dev": "tsup --watch",
    "build": "tsup",
    "storybook": "storybook dev -p 6007",
    "storybook:build": "storybook build",
    "storybook:test": "test-storybook",
    "storybook:test:ci": "concurrently -k -s first -n \"SB,TEST\" -c \"magenta,blue\" \"pnpm storybook:build && npx http-server storybook-static --port 6006 --silent\" \"wait-on http://127.0.0.1:6006 && pnpm storybook:test\"",
    "format": "prettier .",
    "format:fix": "prettier --write .",
    "lint": "eslint .",
    "lint:fix": "eslint --fix .",
    "fix": "npm run format:fix && npm run lint:fix"
  },
  "dependencies": {
    "@floating-ui/dom": "^1.6.10",
    "@radix-ui/react-context-menu": "^2.1.5",
    "@radix-ui/react-dropdown-menu": "^2.0.6",
    "@radix-ui/react-menubar": "^1.0.4",
    "@radix-ui/react-select": "^2.0.0",
    "@radix-ui/react-toast": "^1.1.5",
    "@toshusai/cmpui-core": "link:../core",
    "@toshusai/cmpui-css": "link:../css",
    "react-transition-group": "^4.4.5"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@playwright/test": "^1.41.2",
    "@storybook/addon-essentials": "8.2.6",
    "@storybook/addon-interactions": "8.2.6",
    "@storybook/addon-links": "8.2.6",
    "@storybook/addon-mdx-gfm": "8.2.6",
    "@storybook/addon-onboarding": "8.2.6",
    "@storybook/addon-themes": "8.2.6",
    "@storybook/blocks": "8.2.6",
    "@storybook/react": "8.2.6",
    "@storybook/react-vite": "8.2.6",
    "@storybook/test": "8.2.6",
    "@storybook/test-runner": "^0.17.0",
    "@storybook/theming": "8.2.6",
    "@tabler/icons-react": "^3.15.0",
    "@types/jest": "^29.5.11",
    "@types/react": "^18.2.37",
    "@types/react-dom": "^18.2.15",
    "@types/react-transition-group": "^4.4.11",
    "concurrently": "^8.2.2",
    "eslint": "^9.9.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-import": "^2.29.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "storybook": "8.2.6",
    "tsup": "^8.0.1",
    "typescript": "^5.2.2",
    "vitest": "^1.1.0",
    "wait-on": "^7.2.0"
  },
  "keywords": [
    "react",
    "typescript",
    "components",
    "ui",
    "gui"
  ],
  "author": "toshusai",
  "license": "MIT"
}
