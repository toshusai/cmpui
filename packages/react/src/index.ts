export * from "./components/Button";
export * from "./components/CanvasView";
export * from "./components/Checkbox";
export * from "./components/Circle";
export * from "./components/ColorInput";
export * from "./components/ColorLoupe";
export * from "./components/ColorPickerCircle";
export * from "./components/ColorPickerRect";
export * from "./components/ColorSlider";
export * from "./components/ContextMenu";
export * from "./components/FloatBox";
export * from "./components/Grid";
export * from "./components/HueCircle";
export * from "./components/IconButton";
export * from "./components/IconButton/IconButtonGroup";
export * from "./components/Line";
export * from "./components/ListItem";
export * from "./components/MenuBar/MenuBar";
export * from "./components/MenuBar/MenuBarButton";
export * from "./components/Modal";
export * from "./components/Popover";
export * from "./components/RectGizmo";
export * from "./components/Ruler";
export * from "./components/Select";
export * from "./components/SelectRect";
export * from "./components/Slider";
export * from "./components/SliderNumberField";
export * from "./components/SplitPane";
export * from "./components/SVPicker";
export * from "./components/Switch";
export * from "./components/TextArea";
export * from "./components/TextInput";
export * from "./components/Toast";
export * from "./components/Tooltip";
export * from "./components/TreeView";
export * from "./utils";
