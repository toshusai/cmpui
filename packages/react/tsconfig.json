{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "ESNext",
    "skipLibCheck": true,

    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "emitDeclarationOnly": true,
    "declaration": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "react-jsx",
    "outDir": "dist",

    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,
    "typeRoots": ["node_modules/@types", ".storybook/@types"]
  },
  "include": ["src"]
}
