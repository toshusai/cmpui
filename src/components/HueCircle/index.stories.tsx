import type { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import { HueCircle } from ".";

const meta: Meta<typeof HueCircle> = {
  title: "Colors/HueCircle",
  component: HueCircle,
};

export default meta;
type Story = StoryObj<typeof HueCircle>;

export const Default: Story = {
  render: function Render() {
    const [hue, setHue] = useState(0);

    return <HueCircle hue={hue} onHueChange={setHue} />;
  },
};

export const Size: Story = {
  render: function Render() {
    const [hue, setHue] = useState(0);

    return <HueCircle hue={hue} onHueChange={setHue} size={64} />;
  },
};

export const ForMobile: Story = {
  render: function Render() {
    const [hue, setHue] = useState(0);

    return (
      <HueCircle hue={hue} showLoupe onHueChange={setHue} strokeWidth={32} />
    );
  },
};
