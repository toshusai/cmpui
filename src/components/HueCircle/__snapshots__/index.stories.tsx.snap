// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Colors/HueCircle Default smoke-test 1`] = `
<div class="cmpui_hue-circle__root"
     tabindex="0"
     style="--cmpui-width: 24px; --cmpui-size: 192px;"
>
  <div class="cmpui_hue-circle__hue"
       style="width: 192px; height: 192px;"
  >
  </div>
  <div x="96"
       y="12"
       class="cmpui_circle__root cmpui_hue-circle__nob"
       style="--cmpui-circle-nob-fill: rgb(255,0,0); --cmpui-circle-nob-radius: 8px; --cmpui-circle-nob-stroke-color: rgb(255,255,255); --cmpui-circle-nob-stroke-width: 2px; border: 2px solid rgb(255, 255, 255); left: 88px; top: 4px;"
  >
  </div>
  <div class="cmpui_hue-circle__inner">
  </div>
</div>
`;

exports[`Components/Colors/HueCircle ForMobile smoke-test 1`] = `
<div class="cmpui_hue-circle__root"
     tabindex="0"
     style="--cmpui-width: 32px; --cmpui-size: 192px;"
>
  <div class="cmpui_hue-circle__hue"
       style="width: 192px; height: 192px;"
  >
  </div>
  <div x="96"
       y="16"
       class="cmpui_circle__root cmpui_hue-circle__nob"
       style="--cmpui-circle-nob-fill: rgb(255,0,0); --cmpui-circle-nob-radius: 10.666666666666666px; --cmpui-circle-nob-stroke-color: rgb(255,255,255); --cmpui-circle-nob-stroke-width: 2px; border: 2px solid rgb(255, 255, 255); left: 85.3333px; top: 5.33333px;"
  >
  </div>
  <div class="cmpui_hue-circle__inner">
  </div>
</div>
`;

exports[`Components/Colors/HueCircle Size smoke-test 1`] = `
<div class="cmpui_hue-circle__root"
     tabindex="0"
     style="--cmpui-width: 24px; --cmpui-size: 64px;"
>
  <div class="cmpui_hue-circle__hue"
       style="width: 64px; height: 64px;"
  >
  </div>
  <div x="32"
       y="12"
       class="cmpui_circle__root cmpui_hue-circle__nob"
       style="--cmpui-circle-nob-fill: rgb(255,0,0); --cmpui-circle-nob-radius: 8px; --cmpui-circle-nob-stroke-color: rgb(255,255,255); --cmpui-circle-nob-stroke-width: 2px; border: 2px solid rgb(255, 255, 255); left: 24px; top: 4px;"
  >
  </div>
  <div class="cmpui_hue-circle__inner">
  </div>
</div>
`;
